### ISSUE TITLE
maplegend tool not shown in embedded map

### ISSUE THREAD

**Issue by blackfolder** [oskariorg/oskari-docs#111]
When an embedded map is created then user can select "Show map legend" option. 
![1](https://user-images.githubusercontent.com/32539683/58237366-39f3b880-7d45-11e9-8e7d-adbfe42f2cd8.png)

But when the generated link for map is opened the map legend tool is not shown.
I analysed the process of creating/opening published view and this is what I found (not sure if I understood it right, though).

When Create Embedded Map form is saved, then following JSON data is send to the backend:

`{
  "configuration": {
    "mapfull": {
      "state": {
        "north": 0,
        "east": 0,
        "zoom": 1,
        "srs": "EPSG:3857",
        "selectedLayers": [
          {
            "id": 3,
            "opacity": 100,
            "style": "Basic"
          },
          {
            "id": 6,
            "opacity": 100,
            "style": "default"
          }
        ],
        "plugins": {}
      },
      "conf": {
        "plugins": [
          {
            "id": "Oskari.mapframework.bundle.mapmodule.plugin.ScaleBarPlugin",
            "config": {}
          },
          {
            "id": "Oskari.mapframework.mapmodule.ControlsPlugin",
            "config": null
          },
          {
            "id": "Oskari.mapframework.mapmodule.GetInfoPlugin",
            "config": {
              "colourScheme": {}
            }
          },
          {
            "id": "Oskari.mapframework.bundle.mapmodule.plugin.LogoPlugin",
            "config": {
              "location": {
                "classes": "bottom left"
              }
            }
          }
        ]
      }
    },
    "maplegend": {
      "conf": {},
      "state": {}
    }
  },
  "metadata": {
    "domain": "5",
    "name": "5",
    "language": "en",
    "preview": "fill",
    "toolLayout": "userlayout",
    "style": {
      "font": "arial",
      "toolStyle": "default"
    }
  }
}`

So, "maplegend" tool is not added to to the view's mapfull bundle plugins (AppSetupHandler.js -> setupMapState method). It means that there's no chance that MapLegendPlugin is loaded when the publish view is opened. MapLegendPlugin files are not loaded to browser at all.

What I did to check how I can make things work is:

- I added (hardcoding for now) following code snippet to conf.plugins when the form is saved.

`{
  id: "Oskari.mapframework.bundle.maplegend.plugin.MapLegendPlugin",
  config:
  {
    "location": {
      "classes": "top right"    
    }
  }    
}`
  
![plugin](https://user-images.githubusercontent.com/32539683/58237506-85a66200-7d45-11e9-8ff1-88a48a991a42.png)

- I added required MapLegendPlugin files to oskari\packages\framework\bundle\mapmodule-plugin\bundle.js (just to force loading them).

After that changes maplegend plugin is correctly shown when embedded map is opened. But, of course, this is not a solution for the problem. This should be implemented in a proper way.

My question is: is it an already known problem and is it going to be fixed at some point? If not, could I get any tip what is the right way to fix that? I must admit that I'm not an Oskari pro (yet! :)) so solving this issue is a bit tricky for me decision-wise.

**Comment by ZakarFin** [comment]
It looks like the bundle itself is added in the payload for publishing a map (... "maplegend": { "conf": {}, "state": {} } } ..."  so it's not on the plugins list as most of the other selections. This is produced by https://github.com/oskariorg/oskari-frontend/blob/1.52.0/bundles/framework/maplegend/publisher/MapLegend.js#L96 and merged to the payload that is sent to the server by publisher. 

But according to the screenshot of the appsetup that is loaded -> it's not there. Looking at the appsetup it seems you are using an extremely old Oskari version that still has Openlayers 2 as the mapmodule. Can you check that you don't have an override for this property in oskari-ext.properties: https://github.com/oskariorg/oskari-server/blob/1.52.0/servlet-map/src/main/resources/oskari.properties#L126

**Comment by ZakarFin** [comment]
Or you can try adding that property to oskari-ext.properties if you have some older version of oskari-server installed that didn't have that default value.

**Comment by blackfolder** [comment]
Hi,

Thanks a lot for your time and answers. As I said, I'm still learning Oskari and this time I focused on how the other published view plugins are added and missed adding "simple bundles" in buildPublishedView method. But, at least, I learned a lot while trying to figure out what's going on there. :)

Yes, the app I'm working with uses some old Oskari version (it's been customized so much, that every Oskari merge is very problematic...). I checked Oskari github and our repo history and it seems that during last merge the  actionhandler.AppSetup.bundles.simple property wasn't added to the oskari.properties file (???). I added it there as you suggested and maplegend works completely fine now!
So thanks again for help!
